### PR TITLE
chore: update develop branch to main branch in README.md & docs

### PR DESCRIPTION
**Pull request by karisal-anders**
## Description :sparkles:

**Affects the whole YJDH monorepo**

### chore: update develop branch to main branch in README.md & docs

`develop` branch should not be used for anything anymore, so it is a
good idea to remove references to `develop` branch completely

### CODE REVIEW

File: backend/shared/shared/audit_log/README.md [hunk 1]
```diff
@@ -52,6 +52,6 @@ audit_logging.log(
 ```
 
 Based on:
-- [apartment-application-service audit logging](https://github.com/City-of-Helsinki/apartment-application-service/tree/develop/audit_log)
+- [apartment-application-service audit logging](https://github.com/City-of-Helsinki/apartment-application-service/tree/main/audit_log)
```
Comment (karisal-anders) [inline backend/shared/shared/audit_log/README.md:55]: This is not yjdh monorepo's link but I checked it out and going to the old link here actually redirected to the main branch so it is ok to change it for this repo also

**Comment by karisal-anders** [comment]
As this affects the whole YJDH monorepo I asked for code review from all teams i.e. Helsinki benefit (@sirtawast, @rikuke, @EmiliaMakelaVincit), TET (@tuomas777, @terovirtanen, @arpargo) and Kesäseteli (@nikomakela, @melniiv, @jakezu). Also for the idea of making sure all teams are aware of trying to remove the `develop` branch soon if possible (as a precursor so to speak)—but that's a discussion for another time.

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2784.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2784.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2784.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2784.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2784.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2784.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2784.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2784.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2784.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2784.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2784.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2784.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2784.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2784.dev.hel.ninja :laughing::tada::tada::tada:

**Review by sirtawast — APPROVED** [review]
Nice work! I didn't even realize the links were so off.

**Comment by karisal-anders** [comment]
Even though I asked for code review from all teams, I realize now this is such a small change mostly related to docs that I'll just merge this now as this has been approved by at least someone.

**Pull request merged**